SC_redesign/Crop_and_Soil/soil/layer_data.py

```
@@ -64,6 +64,12 @@ def __post_init__(self):
     @property
     def layer_thickness(self) -> float:
         """thickness of soil layer (mm)"""
+        if self.top_depth < 0:
+            raise ValueError("Top depth of a soil layer cannot be less that 0")
+        elif self.bottom_depth <= 0:
+            raise ValueError("Bottom depth of a soil layer cannot be above or at the top of the soil surface")
+        elif self.top_depth >= self.bottom_depth:
+            raise ValueError("Top depth of a soil layer cannot be at or below the bottom depth of the layer")
```

I don't know how I feel about validation taking place within a `@property`.

@ltp23, can you provide any thoughts on this?

---

Style-wise, I think it's better to raise the errors the moment they occur, so in this case, when the variables become negative somewhere else. We can move those error checks inside the post_init() function, or write a class/static method for checking and call it from post_init(). Ideally, we want the code in property methods to be simple return statements.

---

I will add an issue to address this